series/series.go

```diff
@@ -21,21 +20,27 @@ const (
 )
 
 var (
-	logger     = loggo.GetLogger("juju.juju.series")
-	HostSeries = hostSeries
-
-	seriesOnce sync.Once
-	series     string // filled in by the first call to hostSeries
+	logger = loggo.GetLogger("juju.juju.series")
 )
 
-func hostSeries() string {
-	seriesOnce.Do(func() {
-		var err error
-		series, err = readSeries()
-		if err != nil {
-			panic("unable to determine host series: " + err.Error())
-		}
-	})
+// HostSeries returns the series of the machine the current process is
+// running on.
+//
+// This is not a concurrent-safe operation.
+func HostSeries() (string, error) {
+	series, err := readSeries()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	return series, err
+}
+
+// MustHostSeries calls HostSeries and panics if there is an error.
```

Comment:
There are several places in Juju that currently rely on this behavior panicing. I don't plan on tackling that issue other than re-pointing those areas to this function.